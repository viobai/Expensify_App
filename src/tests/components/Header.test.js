// render only testing component
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../../components/Header';

test('should render Header correctly', () => {
	const wrapper = shallow(<Header />);
	expect(toJson(wrapper)).toMatchSnapshot();
	//const renderer = new ReactShallowRenderer();
	//renderer.render(<Header />);
	//expect(renderer.getRenderOutput()).toMatchSnapshot();
});