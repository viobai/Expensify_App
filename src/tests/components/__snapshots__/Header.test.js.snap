// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Expensify
</h1>,
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          to="/dashboard"
>
          Dashboard
</NavLink>,
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          to="/create"
>
          Create Expense
</NavLink>,
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          to="/help"
>
          Help
</NavLink>,
        <button
          onClick={undefined}
>
          Logout
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expensify",
        },
        "ref": null,
        "rendered": "Expensify",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "is-active",
          "ariaCurrent": "true",
          "children": "Dashboard",
          "to": "/dashboard",
        },
        "ref": null,
        "rendered": "Dashboard",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "is-active",
          "ariaCurrent": "true",
          "children": "Create Expense",
          "to": "/create",
        },
        "ref": null,
        "rendered": "Create Expense",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "is-active",
          "ariaCurrent": "true",
          "children": "Help",
          "to": "/help",
        },
        "ref": null,
        "rendered": "Help",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Logout",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Logout",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Expensify
</h1>,
          <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            to="/dashboard"
>
            Dashboard
</NavLink>,
          <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            to="/create"
>
            Create Expense
</NavLink>,
          <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            to="/help"
>
            Help
</NavLink>,
          <button
            onClick={undefined}
>
            Logout
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expensify",
          },
          "ref": null,
          "rendered": "Expensify",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "ariaCurrent": "true",
            "children": "Dashboard",
            "to": "/dashboard",
          },
          "ref": null,
          "rendered": "Dashboard",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "ariaCurrent": "true",
            "children": "Create Expense",
            "to": "/create",
          },
          "ref": null,
          "rendered": "Create Expense",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "ariaCurrent": "true",
            "children": "Help",
            "to": "/help",
          },
          "ref": null,
          "rendered": "Help",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Logout",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Logout",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
